### PR TITLE
Generate test failures log file for bash

### PR DESCRIPTION
**Pull request by MikeFultonDev**
write check failures to $1/$2_check_failures.log so that jenkins can retain it (and dev can see it)

### CODE REVIEW

File: buildenv [hunk 1]
```diff
@@ -12,11 +12,13 @@ zopen_check_results()
...
+  failures=$(echo "${failuretests}" | wc -l)
   total=$(/bin/find tests/ -name "*.output" | wc -l)
 
-  # Echo the following information to guage build health
+  # Echo the following information to gauge build health
+  echo "${failuretests}" >"$1/$2_check_failures.log"
   echo "actualFailures:$failures"
   echo "totalTests:$total"
-  echo "expectedFailures:14"
+  echo "expectedFailures:15"
```
Comment (IgorTodorovskiIBM) [inline buildenv:23]: Curious, on which system do you get 15 failures? On Jenkins, we get 14 and on zoscan2b, I'm getting 12.

Comment (MikeFultonDev) [reply buildenv:23]: Zoscan2b

Comment (IgorTodorovskiIBM) [reply buildenv:23]: We should add grep as a dependency. I noticed grep and egrep are used in the tests. Maybe that is why we're getting differences?

Comment (MikeFultonDev) [reply buildenv:23]: Will do

Comment (MikeFultonDev) [reply buildenv:23]: still have 15 errors here when adding grep. @IgorTodorovskiIBM  Can you run on zoscan2b to see if it's due to the system or whether it's part of my configuration?

Comment (MikeFultonDev) [reply buildenv:23]: this is my failures list:
```
cat log/20230129_213052_check_failures.log
tests/heredoc.output
tests/read.output
tests/redir.output
tests/test.output
tests/glob.output
tests/printf.output
tests/trap.output
tests/varenv.output
tests/vredir.output
tests/exec.output
tests/builtins.output
tests/lastpipe.output
tests/array.output
tests/histexp.output
tests/intl.output
```

Comment (MikeFultonDev) [reply buildenv:23]: Igor's list:
```
tests/heredoc.output
tests/read.output
tests/redir.output
tests/test.output
tests/glob.output
tests/printf.output
tests/trap.output
tests/varenv.output
tests/vredir.output
tests/exec.output
tests/builtins.output
tests/array.output
```

So 'lastpipe, intl' are the ones that work for Igor and not me.

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM

**Pull request merged**